sensor/common/sensor/sensor.go

```diff
@@ -299,6 +299,7 @@ func (s *Sensor) Start() {
 	if features.PreventSensorRestartOnDisconnect.Enabled() {
 		log.Info("Running Sensor with connection retry: preventing sensor restart on disconnect")
 		go s.communicationWithCentralWithRetries(&centralReachable)
+		// There is no guarantee that the connection to central will be ready when this go routine finishes!
```

Comment:
This func `communicationWithCentralWithRetries` is kind of a special thing :) 
It does 3 things:
- checks the status to maybe change online/offline mode
- triggers the connection
- waits for connection to be closed (and then retries) or waits for a stop signal (and then does not retry).

So this
>  If the routine finishes, then we know for sure the connection is stopped as it only stops if sensor stops.

is not 100% correct, cause we would need to check the signals additionally to tell whether that is an intended or unintended stop (for the former we do not try to reconnect, but for the latter we do retry).

Reply:
> is not 100% correct, cause we would need to check the signals additionally to tell whether that is an intended or unintended stop (for the former we do not try to reconnect, but for the latter we do retry).

🤔 Maybe I misunderstood something but AFAIU the routine can only finish if the `stop` signal is triggered. If that happens, we call `centralCommunication.Stop` before exiting the routine. So if the routine has ended two things have/are happened/happening: Sensor is stopping and the connection with Central is stopped. If my understanding is correct this:

```golang
// There is no guarantee that the connection to central will be ready when this go routine finishes!
```

is not correct because when the go routine finishes, it stops the connection.

I think we might be saying the same thing, but I think the comment is kind of confusing because the way I'd read it without knowing how `communicationWithCentralWithRetries` works is: this go routine is meant to finish when the connection is established or failed. When in reality it does not stop (unless sensor stops).

Reply:
Yeah, now I see that I was wrong. Of course the retry loop can finish in the 3 situations that I listed above, but the goroutine finishes only when we stop retrying and that happens only when we intentionally call `Stop`.

I will adapt the comment to make it correct.